mlir/lib/IR/AffineExpr.cpp

```diff
@@ -1385,7 +1385,7 @@ LogicalResult SimpleAffineExprFlattener::visitModExpr(AffineBinaryOpExpr expr) {
     lhs[getLocalVarStartIndex() + numLocals - 1] = -rhsConst;
   } else {
     // Reuse the existing local id.
-    lhs[getLocalVarStartIndex() + loc] = -rhsConst;
+    lhs[getLocalVarStartIndex() + loc] -= rhsConst;
```

Comment:
This looked right to me before. Can you add a proper test showing that this is wrong?

Reply:
Actually, I printed it and I see what is happening here. Your change makes sense.